implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -277,8 +283,13 @@ public Uni<SmallRyeHealth> getWellnessAsync() {
     public Uni<SmallRyeHealth> getHealthGroupAsync(String groupName) {
         List<Uni<HealthCheckResponse>> checks = new ArrayList<>();
         if (allHealthChecks != null && allAsyncHealthChecks != null) {
-            initUnis(checks, allHealthChecks.select(HealthGroup.Literal.of(groupName)),
-                    allAsyncHealthChecks.select(HealthGroup.Literal.of(groupName)));
+            if (groupName.equals(defaultHealthGroup)) {
+                initUnis(checks, getHealthChecksWithoutHealthGroup(HealthCheck.class),
```

Comment:
But in this case, if you have health checks that have an explicit `@HealthGroup` with the value equal to the name of the default group, they won't be returned, right? Shouldn't we include them too, along with those that don't have an explicit group defined?